### PR TITLE
gh-112334: Regression test that vfork is used when expected.

### PR DESCRIPTION
**Pull request by gpshead**
Adds a regression test #112334. This is the regression test I was pondering in #112617.

This also ensures strace is present in our CI GH (It may have already been there, ubuntu-standard includes it, now we're explicit about the need). I also made sure to install it on some Linux buildbots that I maintain. From my testing, some clearly do not have it.

If this test ever proves a challenge to maintain, further restricting it to a specific platform is fine. The way it works depends on details that can vary between different Linux platforms. ex: aarch64 uses more modern syscalls than x86_64 for the same task.

* Issue: gh-112334

### CODE REVIEW

**Comment by gpshead** [comment]
!buildbot .*root

**Comment by gpshead** [comment]
!buildbot .*root

**Comment by gpshead** [comment]
!buildbot .*aarch64

**Comment by gpshead** [comment]
!buildbot aarch

File: Lib/test/support/script_helper.py [hunk 1]
```diff
@@ -92,13 +92,28 @@ def fail(self, cmd_line):
...
+
+    If _run_using_command= is supplied, it must be a list of
+    command line arguments to prepend to the command line used.
+    Useful when you want to run another command that should launch the
+    python interpreter via its own arguments. ["/bin/echo", "--"] for
+    example could print the unquoted python command line instead of
+    run it.
+    """
     env_required = interpreter_requires_environment()
+    run_using_command = env_vars.pop('_run_using_command', None)
```
Comment (serhiy-storchaka) [inline Lib/test/support/script_helper.py]: Perhaps use a double underscore for consistency with other arguments (`__cwd`, `__isolated`)?

File: Lib/test/test_subprocess.py [hunk 2]
```diff
@@ -3360,6 +3345,89 @@ def exit_handler():
...
+                    try:
+                        subprocess.check_call(
+                                [{true_binary!r}], **dict({sp_kwarg}))
+                    except PermissionError:
+                        if not {expect_permission_error}:
+                            raise"""),
+                    _run_using_command=strace_command,
+                )
+                # Ensure neither vfork() or clone(..., flags=...|CLONE_VFORK|...).
+                self.assertNotRegex(non_vfork_result.err, br"(?mi)vfork")
```
Comment (serhiy-storchaka) [inline Lib/test/test_subprocess.py]: Is `?m` needed here?

File: Lib/test/test_subprocess.py [hunk 3]
```diff
@@ -3360,6 +3345,89 @@ def exit_handler():
...
+        with self.subTest(name="default_is_vfork"):
+            vfork_result = assert_python_ok(
+                    "-c",
+                    textwrap.dedent(f"""\
+                    import subprocess
+                    subprocess.check_call([{true_binary!r}])"""),
+                    _run_using_command=strace_command,
+            )
+            # Match both vfork() and clone(..., flags=...|CLONE_VFORK|...)
+            self.assertRegex(vfork_result.err, br"(?mi)vfork")
```
Comment (serhiy-storchaka) [inline Lib/test/test_subprocess.py]: Is `?m` needed here?

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Pull request merged**